js/integration/knockout/template.js

```diff
@@ -33,9 +34,9 @@ var KoTemplate = TemplateBase.inherit({
     _prepareDataForContainer: function(data, container) {
         if(container && container.length) {
             var containerElement = container.get(0);
-            var containerContext = ko.contextFor(containerElement);
-
-            data = data !== undefined ? data : ko.dataFor(containerElement) || {};
+            var node = getClosestNodeWithContext(containerElement);
+            var containerContext = ko.contextFor(node);
+            data = data !== undefined ? data : ko.dataFor(node) || {};
```

Comment:
NOTE: for experiment with [skip ci]:
```suggestion
            data = data === undefined ? ko.dataFor(node) : data || {};
```